convert_osm/src/split_ways.rs

```diff
@@ -46,10 +46,11 @@ pub fn split_up_roads(
             let count = counts_per_pt.inc(pt);
 
             // All start and endpoints of ways are also intersections.
+            #[allow(clippy::collapsible_if)]
             if count == 2 || idx == 0 || idx == r.center_points.len() - 1 {
-                if !pt_to_intersection.contains_key(&pt) {
+                if let std::collections::hash_map::Entry::Vacant(e) = pt_to_intersection.entry(pt) {
```

Comment:
Less readable IMO, but avoiding the double hashmap lookup is nice.

Reply:
Yeah, the performance should be strictly better. I'll add `use std::collections::hash_map::Entry` to make it more readable.